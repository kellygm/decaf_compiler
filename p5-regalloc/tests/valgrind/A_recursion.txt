==2923496== Memcheck, a memory error detector
==2923496== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==2923496== Using Valgrind-3.22.0 and LibVEX; rerun with -h for copyright info
==2923496== Command: ../decaf inputs/fib.decaf
==2923496== 
fib:
  push BP
  i2i SP => BP
  addI SP, -8 => SP
  loadAI [BP+16] => R0
  loadI 1 => R1
  cmp_LE R0, R1 => R0
  cbr R0 => l1, l2
l1:
  loadAI [BP+16] => R0
  i2i R0 => RET
  jump l0
l2:
  loadAI [BP+16] => R0
  loadI 1 => R1
  sub R0, R1 => R0
  push R0
  call fib
  addI SP, 8 => SP
  i2i RET => R0
  loadAI [BP+16] => R1
  loadI 2 => R2
  sub R1, R2 => R1
  push R1
  storeAI R0 => [BP-8]
  call fib
  addI SP, 8 => SP
  i2i RET => R0
  loadAI [BP-8] => R1
  add R1, R0 => R0
  i2i R0 => RET
  jump l0
l0:
  i2i BP => SP
  pop BP
  return
main:
  push BP
  i2i SP => BP
  addI SP, -8 => SP
  loadI 4 => R0
  storeAI R0 => [BP-8]
  loadI 4 => R0
  push R0
  call fib
  addI SP, 8 => SP
  i2i RET => R0
  i2i R0 => RET
  jump l3
l3:
  i2i BP => SP
  pop BP
  return
RETURN VALUE = 3
==2923496== 
==2923496== HEAP SUMMARY:
==2923496==     in use at exit: 0 bytes in 0 blocks
==2923496==   total heap usage: 2,418 allocs, 2,418 frees, 806,684 bytes allocated
==2923496== 
==2923496== All heap blocks were freed -- no leaks are possible
==2923496== 
==2923496== For lists of detected and suppressed errors, rerun with: -s
==2923496== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
