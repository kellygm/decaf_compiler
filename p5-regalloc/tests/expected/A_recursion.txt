fib:
  push BP
  i2i SP => BP
  addI SP, -8 => SP
  loadAI [BP+16] => R0
  loadI 1 => R1
  cmp_LE R0, R1 => R0
  cbr R0 => l1, l2
l1:
  loadAI [BP+16] => R0
  i2i R0 => RET
  jump l0
l2:
  loadAI [BP+16] => R0
  loadI 1 => R1
  sub R0, R1 => R0
  push R0
  call fib
  addI SP, 8 => SP
  i2i RET => R0
  loadAI [BP+16] => R1
  loadI 2 => R2
  sub R1, R2 => R1
  push R1
  storeAI R0 => [BP-8]
  call fib
  addI SP, 8 => SP
  i2i RET => R0
  loadAI [BP-8] => R1
  add R1, R0 => R0
  i2i R0 => RET
  jump l0
l0:
  i2i BP => SP
  pop BP
  return
main:
  push BP
  i2i SP => BP
  addI SP, -8 => SP
  loadI 4 => R0
  storeAI R0 => [BP-8]
  loadI 4 => R0
  push R0
  call fib
  addI SP, 8 => SP
  i2i RET => R0
  i2i R0 => RET
  jump l3
l3:
  i2i BP => SP
  pop BP
  return
RETURN VALUE = 3
